src/neoxp/Commands/BatchCommand.cs

```diff
@@ -122,6 +123,20 @@ internal async Task ExecuteAsync(IDirectoryInfo root, ReadOnlyMemory<string> com
                                 cmd.Model.Force).ConfigureAwait(false);
                             break;
                         }
+                    case CommandLineApplication<BatchFileCommands.Contract.Download> cmd:
+                       {
+                        var expressNode = chainManager.GetExpressNode();
```

Comment:
nit: indentation

Reply:
Also, txExec already an expressNode instance, so you shouldn't call `GetExpressNode` here. we should probably a wrapper `ContractDownloadAsync` method to `TransactionExecutor`

Reply:
You previously (https://github.com/neo-project/neo-express/pull/209#discussion_r830643990) said
> Obviously, download modifications are not transactions, so TransactionExecutor would not be a good place to put download code shared between direct and batch versions.

Now that you mentioned `nodes` I actually thought of 2 possible issues
1. I think the current code currently only works on a single node chain
2. Clients connected to our consensus node won't know the storage changed because it is not done through a transaction in a block that is relayed over the network :thinking: 

   This might actually be hard to solve. First thing that comes to mind is creating a transaction script that does all the modifications, but only the [contract itself](https://github.com/neo-project/neo/blob/2a64c1cc809d1ff4b3a573c7c22bffbbf69a738b/src/neo/SmartContract/ApplicationEngine.Storage.cs#L79) can modify it's own storage otherwise we'd have a security issue. I have to think more about this

Reply:
I'm OK if this feature only works on single node chains. We just need to add a check at the beginning to make sure that we're on a single node chain and throwing if we're not.

As far as I can tell, there's no way we'll be able to replicate the updated storage to other  non-consensus nodes connected to the blockchain. I'm not sure how common that scenario is tho.

Reply:
Also <sigh> throwing my own words back to me... 

We don't want to create another instance of expressNode. So we can either ignore my previous advice *or* we can create a public property on TransactionExecutor to be able to get the express node instance. Your call

Reply:
1. Regarding single vs multi-node support. Can't we just call download and persist contract in a loop for all consensus nodes? Worst case we need to stop the nodes for a moment to update and then restart them. I don't know if batch mode is actually limited to single node?

2. I think we might be able to solve the chain replication using a [P2P plugin](
https://github.com/neo-project/neo/blob/2a64c1cc809d1ff4b3a573c7c22bffbbf69a738b/src/neo/Plugins/IP2PPlugin.cs#L26). We can put the contractstate + storage in a custom payload, wrap that in an [ExtensiblePayload](https://github.com/neo-project/neo/blob/2a64c1cc809d1ff4b3a573c7c22bffbbf69a738b/src/neo/Network/P2P/Payloads/ExtensiblePayload.cs). You can then still run a normal client, it just needs the plugin to handle the payload.

   I think we should handle this in a separate PR.

Reply:
BTW, `TransactionExecutor` already has a `ExpressNode` property. `checkpoint create` and `fastfwd` commands in batch mode both use the `ExpressNode` property already. `contract download` can use it to.

We could totally look at doing something like download and persist in a loop. That might work for checkpoints, which are currently limited to single node chains. I'll think about that and the P2P plugin idea some more today.